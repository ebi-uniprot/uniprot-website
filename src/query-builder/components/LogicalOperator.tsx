import React from 'react';
import { Operator } from '../types/searchTypes';

const operators: Operator[] = [Operator.AND, Operator.OR, Operator.NOT];

const LogicalOperator = ({
  value,
  handleChange,
}: {
  value: Operator;
  handleChange: (value: Operator) => void;
}) => (
  <select
    className="query-builder__logic"
    value={value}
    data-testid="query-builder-logic-select"
    onChange={(e) => handleChange(e.target.value as Operator)}
  >
    {operators.map((op) => (
      <option value={op} key={op}>
        {op}
      </option>
    ))}
  </select>
);

export default LogicalOperator;
