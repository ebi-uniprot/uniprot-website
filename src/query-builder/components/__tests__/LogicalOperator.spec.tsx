import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LogicalOperator from '../../../query-builder/components/LogicalOperator';

let rendered;
let props;

describe('LogicalOperator component', () => {
  beforeEach(() => {
    props = {
      value: 'AND',
      handleChange: jest.fn(),
    };
    rendered = render(<LogicalOperator {...props} />);
  });

  test('should change evidence', () => {
    const { getByTestId } = rendered;
    const select = getByTestId('query-builder-logic-select');
    fireEvent.change(select, {
      target: { value: 'OR' },
    });
    expect(props.handleChange).toBeCalled();
  });

  test('should render', () => {
    const { asFragment } = rendered;
    expect(asFragment()).toMatchSnapshot();
  });
});
