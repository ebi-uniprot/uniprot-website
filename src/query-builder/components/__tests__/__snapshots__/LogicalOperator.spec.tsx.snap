// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogicalOperator component should render 1`] = `
<DocumentFragment>
  <select
    class="query-builder__logic"
    data-testid="query-builder-logic-select"
  >
    <option
      value="AND"
    >
      AND
    </option>
    <option
      value="OR"
    >
      OR
    </option>
    <option
      value="NOT"
    >
      NOT
    </option>
  </select>
</DocumentFragment>
`;
